index.d.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Decides whether or not the document is reader-able without parsing the whole thing.
+ * @return {boolean} Whether or not we suspect Readability.parse() will suceeed at returning an article object.
+ */
+export function isProbablyReaderable(
+  document: Document,
+  options?: {
+    /** The minimum node content length used to decide if the document is readerable. */
+    minContentLength?: number;
+    /** The minumum cumulated 'score' used to determine if the document is readerable. */
+    minScore?: number;
+    /** The function used to determine if a node is visible. */
+    visibilityChecker?: (node: Node) => boolean;
+  }
+): boolean;
+
+export class Readability {
+  constructor(
+    document: Document,
+    options?: {
+      debug?: boolean;
+      maxElemsToParse?: number;
+      nbTopCandidates?: number;
+      charThreshold?: number;
+      classesToPreserve?: string[];
+      keepClasses?: boolean;
+      serializer?: (node: Node) => string;
```

Comment:
The `serializer` option is used exclusively to obtain the `content` property in the `parse()` call. It's not necessary for it to produce a string. Whatever you return from the serializer function will be reflected in `content`, but I'm not sure how to model that in TypeScript.

Reply:
Oh right, thanks. https://github.com/mozilla/readability/pull/667/commits/44a9c32bc407e2688cb521568163169a72ccba62 should do it